app/code/Magento/CatalogImportExport/Model/Import/Product.php

```diff
@@ -1737,7 +1749,43 @@ protected function _saveProducts()
                 $position = 0;
                 foreach ($rowImages as $column => $columnImages) {
                     foreach ($columnImages as $columnImageKey => $columnImage) {
-                        if (!isset($uploadedImages[$columnImage])) {
+                        $hash = '';
+                        if (filter_var($columnImage, FILTER_VALIDATE_URL) === false) {
+                            $filename = $importDir . DIRECTORY_SEPARATOR . $columnImage;
+                            if ($this->fileDriver->isExists($filename)) {
+                                $hash = hash_file(
+                                    'sha256',
+                                    $importDir . DIRECTORY_SEPARATOR . $columnImage
+                                );
+                            }
+                        } else {
+                            $hash = hash_file('sha256', $columnImage);
```

Comment:
We probably don't need to call the hash_file twice here, wouldn't be better to define something like $imageFilePath and then call the private method to hash the file just once.